syncstorage-spanner/src/metadata.rs

```diff
@@ -28,7 +28,7 @@ const ROUTING_KEY: &str = "x-goog-request-params";
 const LEADER_AWARE_KEY: &str = "x-goog-spanner-route-to-leader";
 
 /// Our user agent value for [METRICS_KEY]
-const USER_AGENT: &str = "gcp-grpc-rs";
+const USER_AGENT: &str = "gl-external/{version} gccl/{version}";
```

Comment:
not super critical but we might as well only do this once: we could use `lazy_static` but the new `OnceLock` [e.g.](https://github.com/mozilla-services/autopush-rs/blob/master/autopush-common/src/logging.rs#L13) is probably preferable to it at this point.

Though in practice `OnceLock`'s API is also a bit funny for this (because you either need to move this `format!` looking constant string into a runtime code block or maybe put the entire `OnceLock` into the runtime code block instead, keeping the string template here as const module level).

So maybe concat! is the simplest:

```suggestion
const USER_AGENT: &str = concat!(
    "gl-external/",
    env!("CARGO_PKG_VERSION"),
    " gccl/",
    env!("CARGO_PKG_VERSION")
);
```

Reply:
Sounds good. I also built https://github.com/mozilla-services/google-cloud-rust/pull/38 which exposes the VERSION of the actual rust library. I may fix up this one to use it instead.

Reply:
Probably want to hold off merging this until we get an authoritative answer from Google about some of the version number questions.

Reply:
This is no longer valid (or needed)